### DISCUSSION TITLE
MongoDB Authenticator types

### DISCUSSION THREAD

**Discussion by gregorvand** [MasterKale/SimpleWebAuthn#375] [Troubleshooting]
Is anyone using MongoDB with this library?
Hitting a bit of a wall with which type to store `credentialID` so that when `generateAuthenticatorOptions` does the following, A value is actually populated.

```
allowCredentials: allowCredentials === null || allowCredentials === void 0 ? void 0 : allowCredentials.map(cred => ({
            ...cred,
            id: iso_1.isoBase64URL.fromBuffer(cred.id),  <<< drops 
        })),
```
I have debugged this to the point that the `cred.id` definitely makes it into the function, and is dropped every time by the above. Any thoughts? I can only pin this down to how it is being stored in the db.

**Comment by MasterKale** [comment]
Hello @gregorvand, what is the value of `cred.id` that's getting fed into `isoBase64URL.fromBuffer()`?

**Comment by gregorvand** [comment]
Thanks @MasterKale , turns out this was a side effect of not having a `transports` value - rather than an error, it just returned an empty authenticator `id` which probably 'makes sense', it was just not obvious passing in this albeit optional value as null would have this effect.

All working now - for future reference to any MongoDB users, the main update I had to make to the sample code was in the `verifyAuthenticationResponse` to deal with the data types correctly:


```
import base64url from 'base64url';
...

let verification;
try {
  verification = await verifyAuthenticationResponse({
    response: body.asseResp,
    expectedChallenge,
    expectedOrigin: 'http://localhost:5173',
    expectedRPID: 'localhost',
    authenticator: {
      credentialPublicKey: authenticator.credentialPublicKey.buffer,  <<<
      credentialID: base64url.toBuffer(authenticator.credentialID), <<<
      counter: authenticator.counter, 
  },
  });
} catch...
```


This is using the doc's recommendations to store `credentialID` as base64url-encoded `String` type, and `credentialPublicKey` as `Binary` (default of MongoDB if adding without predefined schema)

And thanks again @MasterKale for all of your work on this library and documentation!
